homotopy-graphics/src/style.rs

```diff
@@ -1,15 +1,26 @@
+use std::fmt::Write;
+
 use homotopy_core::Generator;
 
+use crate::tikz;
+
 pub trait GeneratorStyle {
+    type Color: RenderableColor;
+
     fn label(&self) -> Option<String>;
     fn shape(&self) -> Option<VertexShape>;
-    // TODO(thud): migrate color-related code to here for consistency
+    fn color_point(&self) -> Self::Color;
```

Comment:
I reckon this should contain a single `color` function.

Reply:
As mentioned in one of the other comments, I think this can be fixed by moving the `palette` dependency from `homtopy_web` into `homotopy_graphics`.